docs/guidelines/development.rst

```diff
@@ -7,15 +7,68 @@ This documents proposes development guidelines to increase work efficiency and s
 
 **NOTE**: All the proposal here are the results of authors' personal experiences. Saying that, if you have any idea to make them better you are very wellcome to create a PR.
 
+Overview
+=========
 
-1. Use *pre-commit* for formatting and linting
-===============================================
+1. `Making changes to the code-base by submitting a Pull Request <#>`_
+
+2. `Using *pre-commit* <#use-pre-commit-for-formatting-and-linting>`_
+
+
+Code Changes and Pull Request Submissions
+==========================================
+
+To make a change of the existing code base, Pull Request (GitHub) or Merge Request (GitLab) are used.
+This section describes in short process with valuable tips to make your and reviewers' life easier.
+
+#. Check with the team if it is usual to submit a PR/MR from a fork or directly into target repository.
+   This usually depends on the team size and organization.
+   When working with public repositories you always need to create a fork.
+
+   .. note::
+
+      In *Stogl Robotics* we have the organization called `StoglRobotics-forks <https://github.com/StoglRobotics-forks>`_ where all forks of public repositories live and are accessible for writing by all team members.
+      This simplify collaboration inside the team - there is no need for individual access grants when using forks under your user.
+      **Always** check if there is already a fork in *StoglRobotics-forks* organization and if not create it.
+
+#. Start development always from the up-to date state of the repositories default branch (usually called *master* or *main* - for simplicity we call it *master* here).
+   Take into account that master branch of your fork is usually not up to date with the upstream repository.
+   Therefore be careful about that and use the opportunity to sync *master* branch of the fork to the state of the *master* branch of the upstream repository.
+
+#. **Always** create a new branch for each feature or bug fix. Don't submit PRs from *master* branch.
+
+#. **Before** starting development check how the branch will be merged, using *merge commit* or *squash* method.
+   If *merge commit* is used make sure that each of your commits is clean and named properly since they will become part of the repository's history.
+
+   .. note::
+
+      In *Stogl Robotics* we are always *squashing* commits, i.e., one future or one bug fix is one commit in the default branch. The commit message is edited before merging.
```

Comment:
`In *Stogl Robotics* we are always *squashing* commits, i.e., one feature or one bug fix is one commit in the default branch. The commit message is edited before merging.`